### PR TITLE
Add save_existing! to Her::Model::ORM

### PR DESCRIPTION
**Pull request by ikuto0608**
Similar to User.save_existing(), except that ResourceInvalid is raised if the save fails.

### CODE REVIEW

**Comment by ikuto0608** [comment]
Hi @edtjones 

There was the situation I didn’t want to use `save/save!` instead.

ex)
```
# External api model with her
class External::User
  attr_reader :hoge
  attr_accessor :fuga
end

user = External::User.find(1)
user.fuga = 'anything' # :fuga is allowed to change

# failed
user.save # -> failed because it sent a PUT request with all elements including `:hoge`,
          # and the external api checks whether the params includes only rewritable params or not,
          # and if the params has read only item, it's not gonna just ignore but `422 UNPROCESSABLE ENTITY`.
          # So I needed to send only the params which I wanted to modify.

# success
External::User.save_existing(user.id, { fuga: 'anything' })
```

Thanks!

File: lib/her/model/orm.rb [hunk 1]
```diff
@@ -223,6 +223,13 @@ def save_existing(id, params)
           resource
         end
 
+        # Similar to User.save_existing(), except that ResourceInvalid is raised if the save fails
```
Comment (zacharywelch) [inline lib/her/model/orm.rb]: Let's go ahead and refactor `save_existing` to use your method

```ruby
def save_existing(id, params)
  save_existing!(id, params)
rescue Her::Errors::ResourceInvalid => e
  e.resource
end
```

Comment (zacharywelch) [inline lib/her/model/orm.rb]: `Similar to .save_existing but raises ResourceInvalid if save fails`

File: spec/model/orm_spec.rb [hunk 2]
```diff
@@ -423,6 +424,15 @@ def friends
       expect(@user.fullname).to eq("Lindsay Fünke")
     end
 
+    it "handle resource update through the .update class method" do
```
Comment (zacharywelch) [inline spec/model/orm_spec.rb]:
```ruby
it "handle resource update through .save_existing!" do
```

File: spec/model/orm_spec.rb [hunk 3]
```diff
@@ -423,6 +424,15 @@ def friends
...
+      @user = Foo::User.save_existing!(1, fullname: "Lindsay Fünke")
+      expect(@user.fullname).to eq("Lindsay Fünke")
+    end
+
+    it "raises ResourceInvalid when #save_existing! gets errors" do
```
Comment (zacharywelch) [inline spec/model/orm_spec.rb]:
```ruby
it "raises ResourceInvalid when .save_existing! has errors" do
```

**Comment by ikuto-rise** [comment]
Thanks @zacharywelch for doing code review, I updated the code!

**Comment by ikuto0608** [comment]
Thanks @zacharywelch for doing code review, I updated the code!

**Pull request merged**